var/spack/repos/builtin/packages/namd/package.py

```diff
@@ -45,6 +46,8 @@ class Namd(MakefilePackage, CudaPackage):
         description="Enables Tcl and/or python interface",
     )
 
+    variant("avxtiles", default=False, description="Enable avxtiles")
```

Comment:
This has no effect when the uarch doesn't support AVX512. I think in principle this:
```
variant("avxtiles", default=False, when="target=x86_64_v4:", description="Enable avxtiles")
```
should work too, even though constraining a variant with a target condition is seldom used. Can you confirm it work correctly?

Reply:
Thanks for the suggestion! We have made the necessary changes!